packages/wrangler/src/pages/publish.tsx

```diff
@@ -373,7 +383,24 @@ export const Handler = async ({
 	 * – this includes its routing and middleware characteristics.
 	 */
 	if (_workerJS) {
-		formData.append("_worker.js", new File([_workerJS], "_worker.js"));
+		let workerFileContents = _workerJS;
+		if (bundleWorker) {
+			const outfile = join(tmpdir(), `./bundledWorker-${Math.random()}.mjs`);
+			await buildRawWorker({
```

Comment:
Do we not need to listen for `onEnd` to complete here and in the `checkWorker()` call?

Reply:
I think `onEnd()` is only really helpful when `watch: true`, right?
Without watching the call to `buildRawWorker()` will only resolve after the build completes.

Reply:
For watch mode, the returned promise resolves after the first build but then we need a way to be notified after subsequent builds.